activesupport/lib/active_support/execution_context.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+module ActiveSupport
+  module ExecutionContext # :nodoc:
+    class << self
+      include ActiveSupport::Callbacks
+      define_callbacks :change
+
+      def after_change(&block)
+        singleton_class.set_callback(:change, :after, &block)
+      end
```

Comment:
Feels a little overwrought to use `Callbacks` here when we're only intending to expose a block based after callback. How about just capturing the blocks directly?

```ruby
def after_change(&block)
  @after_change_callbacks << block
end
```

And then have:

```ruby
def run_after_change_callbacks
  @after_change_callbacks.each(&:call)
end
```

Reply:
> How about just capturing the blocks directly?

Yeah, I used it for consistency with other places like `CurrentAttributes` etc, but on second thought this is for internal use only so we can go with something much simpler. I'll change it.

Reply:
Done.